### ISSUE TITLE
Same rule has different priority in style guide and in eslint-plugin-vue

### ISSUE THREAD

**Issue by ocavue** [vuejs/eslint-plugin-vue#540]
**Please describe what the rule should do:**

Provide a warning when using `v-if` on the same element as `v-for`.

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```html
<PostCard
    v-for="page in pages"
    v-if="isVisiblePost(page)"
    :key="page.key"
    :page="page"
    :showContent="false"
    :showLink="true"
/>
```

```html
<ul>
  <li
    v-for="user in users"
    v-if="shouldShowUsers"
    :key="user.id"
  >
    {{ user.name }}
  <li>
</ul>
```

**Why should this rule be included**

1. [Style guide](https://vuejs.org/v2/style-guide/#Avoid-v-if-with-v-for-essential) has provided some reasons. This rule's priority is essential, which is the same priority as rule [vue/valid-v-for](https://vuejs.org/v2/style-guide/#Keyed-v-for-essential) 
2. Because of unknown reason, I found out that if I use `v-if` with `v-for`, safari will have some wrong behave. I can bypass it if I follow style guide's suggestion (see link above). https://github.com/vuejs/vuepress/issues/678

**Comment by mysticatea** [comment]
Thank you for this issue.

I think you are looking for [no-use-v-if-with-v-for](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/no-use-v-if-with-v-for.md).

**Comment by ocavue** [comment]
@mysticatea 

I found out that vetur didn't show a warning because `no-use-v-if-with-v-for` is a "Priority C: Recommended rule" and vetur's default linting configuration is based on eslint-plugin-vue's "Priority A: Essential" set.

`no-use-v-if-with-v-for`'s has a different priority [in style guide(beta)](https://vuejs.org/v2/style-guide/#Avoid-v-if-with-v-for-essential) and [in eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue#priority-c-recommended-minimizing-arbitrary-choices-and-cognitive-overhead). Should these two projects have the same priority for each rule or their priorities are not related.

**Comment by mysticatea** [comment]
You are right. We should fix it within `5.0.0-beta.x`.

**Comment by ocavue** [comment]
ok, I will reopen this issue and change the title just for reminding.

![image](https://user-images.githubusercontent.com/24715727/43366197-3c57df02-936c-11e8-82fc-4d71186a27d9.png)

**Comment by michalsnik** [comment]
Updated on master
